src/material/core/ripple/ripple-renderer.ts

```diff
@@ -153,16 +153,19 @@ export class RippleRenderer implements EventListenerObject {
     const userTransitionProperty = computedStyles.transitionProperty;
     const userTransitionDuration = computedStyles.transitionDuration;
 
-    // Note: We detect whether animation is forcibly disabled through CSS by the use of
-    // `transition: none`. This is technically unexpected since animations are controlled
-    // through the animation config, but this exists for backwards compatibility. This logic does
-    // not need to be super accurate since it covers some edge cases which can be easily avoided by users.
+    // Note: We detect whether animation is forcibly disabled through CSS (e.g. through
+    // `transition: none` or `display: none`). This is technically unexpected since animations are
+    // controlled through the animation config, but this exists for backwards compatibility. This
+    // logic does not need to be super accurate since it covers some edge cases which can be easily
+    // avoided by users.
     const animationForciblyDisabledThroughCss =
       userTransitionProperty === 'none' ||
       // Note: The canonical unit for serialized CSS `<time>` properties is seconds. Additionally
       // some browsers expand the duration for every property (in our case `opacity` and `transform`).
       userTransitionDuration === '0s' ||
-      userTransitionDuration === '0s, 0s';
+      userTransitionDuration === '0s, 0s' ||
+      // If the container is 0x0, it's likely `display: none`.
+      (containerRect.width === 0 && containerRect.height === 0);
```

Comment:
LGTM, but as said, this logic exists for backwards compatibility. Do we actually want to expand it further, or is there a specific issue we are addressing (internally or externally)?

Reply:
The specific issue is the one I mentioned in the description. It's a bit of an edge case, but I can totally see somebody setting display: none on the ripple container and accidentally causing a memory leak. I sent out the fix since it's pretty cheap for us to guard against it.

Reply:
It was intentionally not supporting all cases back when we added this logic, so I don't know if we necessarily would want to do this. At the same time though it feels like a simple addition- but we should be clear that we do not intend to handle all "force disable cases"- also I think this may not be a fix since working as intended